File: java/src/main/java/io/fivexer/sdk/model/WorkerBreakMetric.java
package io.fivexer.sdk.model;

/**
 * One worker's break totals over a window.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerBreakMetric {
    private String workerId;
    private String label;
    private int count;
    private long totalBreakMs;
    private long longestBreakMs;
    private boolean active;

    public String getWorkerId() { return workerId; }
    public String getLabel() { return label; }
    public int getCount() { return count; }
    public long getTotalBreakMs() { return totalBreakMs; }
    public long getLongestBreakMs() { return longestBreakMs; }
    public boolean isActive() { return active; }
}

File: php/src/Model/WorkerBreakMetric.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * One worker break totals over a window.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerBreakMetric
{
    public function __construct(
        public readonly string $workerId,
        public readonly string $label,
        public readonly int $count,
        public readonly int $totalBreakMs,
        public readonly int $longestBreakMs,
        public readonly bool $active,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workerId: (string) ($data['workerId'] ?? ''),
            label: (string) ($data['label'] ?? ''),
            count: (int) ($data['count'] ?? 0),
            totalBreakMs: (int) ($data['totalBreakMs'] ?? 0),
            longestBreakMs: (int) ($data['longestBreakMs'] ?? 0),
            active: (bool) ($data['active'] ?? false),
        );
    }
}
